src/compiler/emitter.ts

```diff
@@ -2777,34 +2776,61 @@ var __param = this.__param || function(index, decorator) { return function (targ
                 return getCombinedNodeFlags(node.parent);
             }
 
-            function renameNonTopLevelLetAndConst(node: Node): void {
+            function renameBlockScopedValueIfNecessary(node: Node, meaning: SymbolFlags): void {

```

Comment:
I wouldn't call this meaning. I would call it symbolKind or something like that.